### PR TITLE
Write docs for `calyx-pass-explorer`

### PR DESCRIPTION
**Pull request by ethanuppal**
This PR adds a documentation page for the `calyx-pass-explorer` tool, serving as a gentle introduction to its usage as well as an overview of its code.

- I've added `mdbook-callouts` as a way to get Obsidian-style quote blocks. Specifically, I'm using this for the `[!TIP]` environment. I can remove this if we don't want another dependency.
- I did some unrelated things (to `docs/contributors.md`, `docs/github.md`, and `docs/intro.md`) that I can move into another PR if that would be better.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -17,15 +17,17 @@ jobs:
       - name: Setup mdBook
         uses: peaceiris/actions-mdbook@v1
         with:
-          mdbook-version: 'latest'
+          mdbook-version: "latest"
+      - name: Install callouts preprocessor
+        run: cargo install --git https://github.com/ToolmanP/rs-mdbook-callout
```
Comment (rachitnigam) [inline .github/workflows/docs.yml:22]: Is there a way to instead install a specific crate? Also, how does this affect local build? For example, we'd have to tell the users to the documentation system how to install this...

File: docs/github.md [hunk 2]
```diff
@@ -33,6 +36,9 @@ within Rust to suppress the lint.
 If changes are made to the `Dockerfile` then the CI will automatically rebuild
 the Docker image and run your tests on it.
 
+## Local Development
```
Comment (rachitnigam) [inline docs/github.md:39]: This should be before `CI Behaviors` and should point to the installation flow page as well. Something like: "Once you've set up Calyx for contributing <link>, install the git hooks.

If we're recommending `seutp_hooks.sh`, let's make sure it runs `cargo clippy` and `black` on the related files.

File: docs/intro.md [hunk 3]
```diff
@@ -42,29 +44,44 @@ First, install [Rust][rust].
...
+
 ```shell
 /bin/sh setup_hooks.sh
 ```
 
+You can build the docs by installing `mdbook` and the callouts preprocessor:
+
+```sh
+cargo install mdbook 
+cargo install --git https://github.com/ToolmanP/rs-mdbook-callout
```
Comment (rachitnigam) [inline docs/intro.md:75]: Any way to install this in a versioned manner? I'm a little concerned about installing from github.

File: docs/SUMMARY.md [hunk 4]
```diff
@@ -31,6 +31,7 @@
 # Compiler Development Guide
 
 - [The Calyx Compiler](./compiler.md)
+- [Visualizing Compiler Passes](./dev/calyx-pass-explorer.md)
```
Comment (rachitnigam) [inline docs/SUMMARY.md:34]: This should be under the `Tools` section.

File: docs/github.md (hunk 2 above)
Comment (ethanuppal) [reply docs/github.md:39]: Screenshot 2024-11-06 at 10 17 58 AM

I'll move it!

**Comment by ethanuppal** [comment]
I actually have no idea why the interpreter tests aren't passing.

**Comment by rachitnigam** [comment]
Argh, I think this is because they are not sorted deterministically. The results are the same but in different order. Thoughts @ekiwi?

**Comment by ekiwi** [comment]
> Argh, I think this is because they are not sorted deterministically. The results are the same but in different order. Thoughts @ekiwi?
Maybe somehow the JQ stage is not quite doing what we think it should be doing?

**Comment by ethanuppal** [comment]
Screenshot 2024-11-06 at 11 21 39 AM
I don't even have that in my `interp/tests/runt.toml`?

**Comment by ethanuppal** [comment]
Oh I think I should replace on top of main

File: interp/tests/runt.toml [hunk 5]
```diff
@@ -193,9 +193,9 @@ fud2 --from calyx --to jq \
 [[tests]]
 name = "numeric types correctness and parsing"
 paths = [
-  "../../tests/correctness/numeric-types/parsing/*.futil",
-  "../../tests/correctness/numeric-types/bitnum/*.futil",
-  "../../tests/correctness/numeric-types/fixed-point/*.futil",
+    "../../tests/correctness/numeric-types/parsing/*.futil",
```
Comment (ekiwi) [inline interp/tests/runt.toml:196]: this change seems unnecessary

Comment (ethanuppal) [reply interp/tests/runt.toml:196]: I can revert it -- it's my neovim formatting setup.

**Comment by ekiwi** [comment]
@rachitnigam : Is this one ready to go?

**Comment by rachitnigam** [comment]
Yes, LGTM! Thanks for adding this @ethanuppal!

**Pull request merged**